src/components/LoadingIndicator/LoadingIndicator.test.tsx

```diff
@@ -1,48 +1,103 @@
+  // defined
+  test('should render loading indicator container', () => {
+    const { container } = render(<LoadingIndicator {...MOCK_DEFAULT_PROPS} />);
+    expect(container.firstChild).toHaveClass('zui-loading-indicator');
 
-test('should render spinner', () => {
-  const { getByTestId } = renderComponent({});
-  const spinner = getByTestId(TEST_ID.SPINNER);
-  expect(spinner).toBeInTheDocument();
-});
+    const loadingIndicatorContainer = container.getElementsByClassName('zui-loading-indicator');
+    expect(loadingIndicatorContainer.length).toBe(1);
+  });
+
+  test('should render spinner component', () => {
+    const { getByText } = render(<LoadingIndicator {...MOCK_DEFAULT_PROPS} />);
 
-test('should render text', () => {
-  const { getByTestId } = renderComponent({ text: mockText });
-  const text = getByTestId(TEST_ID.TEXT);
-  expect(text).toBeInTheDocument();
-  expect(text).toHaveClass('zui-loading-indicator-text');
-  expect(text).toHaveClass('zui-loading-indicator-text-bottom');
+    expect(getByText(MOCK_SPINNER)).toBeTruthy();
+  });
+
+  test('should render message container and message value', () => {
+    const { container } = render(<LoadingIndicator {...MOCK_DEFAULT_PROPS} />);
+    const textContainer = container.getElementsByClassName('zui-loading-indicator-message');
+
+    expect(textContainer.length).toBe(1);
+    expect(textContainer[0].textContent).toEqual(MOCK_DEFAULT_PROPS.message);
+  });
+
+  // undefined
+  test('should not render message container', () => {
+    const { container } = render(<LoadingIndicator {...MOCK_DEFAULT_PROPS} message={undefined} />);
+    const loadingIndicatorContainer = container.getElementsByClassName('zui-loading-indicator');
+    const messageContainer = container.getElementsByClassName('zui-loading-indicator-message');
+
+    expect(loadingIndicatorContainer).not.toContain(messageContainer);
+    expect(messageContainer.length).toBe(0);
+  });
 });
 
-test('should render text with custom postiion', () => {
-  const { getByTestId } = renderComponent({ text: mockText, spinnerPosition: 'left' });
-  const text = getByTestId(TEST_ID.TEXT);
-  expect(text).toHaveClass('zui-loading-indicator-text-left');
+describe('class names', () => {
+  // defined
+  test('should apply className prop to container class', () => {
+    const { container } = render(<LoadingIndicator {...MOCK_DEFAULT_PROPS} />);
+    const loadingIndicatorContainer = container.getElementsByClassName('zui-loading-indicator');
+
+    expect(loadingIndicatorContainer[0]).toHaveClass(MOCK_DEFAULT_PROPS.className);
+  });
+
+  test('should apply spinnerPosition prop to container class', () => {
+    const { container } = render(<LoadingIndicator {...MOCK_DEFAULT_PROPS} />);
+    const loadingIndicatorContainer = container.getElementsByClassName('zui-loading-indicator');
+
+    expect(loadingIndicatorContainer[0]).toHaveClass(`zui-loading-indicator-${MOCK_DEFAULT_PROPS.spinnerPosition}`);
+  });
```

Comment:
I like to be explicit with the props relevant to the test, e.g.
```
render(<LoadingIndicator {...DEFAULT_PROPS} spinnerPosition={'top'} />)
```
It's a small readability thing that makes the tests a little bit easier to understand. I'm not too attached to this code style though - if you think it has downsides let me know!

Reply:
Roger that, I'm all for that. Updated. 

You'll see I've used an enum for spinnerPosition value. Error throws if I use a hardcoded string like 'top'